bootloader/src/pyi_archive.c

```diff
@@ -377,6 +378,145 @@ findDigitalSignature(ARCHIVE_STATUS * const status)
 #endif /* ifdef _WIN32 */
 }
 
+/*
+ * Try to find the pydata section in an ELF executable.
+ * Returns the end of the section if found.
+ */
+static int
+findElfSection(ARCHIVE_STATUS * const status)
+{
+#if !defined(WIN32) && !defined(__APPLE__)
+    char head[64];
+    uint16_t endianess_bytes = 0x0201;
+    int8_t bits;
+    uint64_t e_shoff;
+    uint16_t e_shentsize;
+    uint16_t e_shnum;
+    uint16_t e_shstrndx;
+    char *section;
+    uint64_t shstr_offset;
+    uint64_t shstr_size;
+    char *shstr;
+    uint32_t section_idx;
+    uint64_t pydata_offset;
+    uint64_t pydata_size;
+
+    fseek(status->fp, 0, SEEK_SET);
+    if (fread(head, 64, 1, status->fp) != 1) {
+        return -1;
+    }
+
+    /* Check if ELF magic present */
+    if (head[0] != 0x7F || head[1] != 'E' || head[2] != 'L' || head[3] != 'F') {
+        VS("LOADER: %s is not an ELF binary\n", status->archivename);
+        return -1;
+    }
+
+    /* Ensure endianess of file matches */
+    if (head[0x05] != *(uint8_t*)&endianess_bytes) {
```

Comment:
This will convert `endianess_bytes` to 1. You don't need to cast if you only use a single byte when you declare the variable.

Reply:
@lkollar Depending on endianess we read 0x01 or 0x02 here and compare with the header value. See also above.

@htgoebel Not sure if it could be an issue on e.g. bi-endian systems. Since it is an inexpensive check, I decided to include it for safety. What do you think?

Reply:
ACK for the test. But is there some symbol instead of using `0x05`? If there is none in a header-file, I suggest defining one ourself. (If there is a header file, you could still consider copying the definition here.)